bin/clang/astVisitor.cc

```diff
@@ -212,12 +212,22 @@ CompilerGlobals::setup(clang::CompilerInstance* CI)
 
 
   if (includeListOpt.getNumOccurrences()){
+    char fullpathBuffer[1024];
     StringRef sref(includeListOpt);
     std::istringstream sstr(sref.str());
     std::string path;
     while (std::getline(sstr, path, ':'))
     {
-      includePaths.push_back(path);
+      //make sure the full paths (following symlinks)
+      //are included for checking
+      const char* fullpath_cstr = realpath(path.c_str(), fullpathBuffer);
+      if (fullpath_cstr){
+        realSystemIncludePaths.push_back(fullpath_cstr);
+      } else {
```

Comment:
Can `realpath` fail while the path is valid?  Part of my original concern I wanted addressed was the question that if `realpath` fails do we just have an error?

Reply:
Yeah, I guess it's probably safer to error out. I haven't encountered a system so far that realpath fails on.